examples/js/controls/OrbitControls.js

```diff
@@ -7,700 +7,740 @@
+		// backwards compatibility
+		this.noZoom = false;
+		this.zoomSpeed = 1.0;
 
-		thetaDelta = 0;
-		phiDelta = 0;
-		scale = 1;
-		pan.set( 0, 0, 0 );
+		// Set to true to disable this control
+		this.noRotate = false;
+		this.rotateSpeed = 1.0;
 
-		// update condition is:
-		// min(camera displacement, camera rotation in radians)^2 > EPS
-		// using small-angle approximation cos(x/2) = 1 - x^2 / 8
+		// Set to true to disable this control
+		this.noPan = false;
+		this.keyPanSpeed = 7.0;	// pixels moved per arrow key push
 
-		if ( lastPosition.distanceToSquared( this.object.position ) > EPS
-		    || 8 * (1 - lastQuaternion.dot(this.object.quaternion)) > EPS ) {
+		// Set to true to automatically rotate around the target
+		this.autoRotate = false;
+		this.autoRotateSpeed = 2.0; // 30 seconds per round when fps is 60
 
-			this.dispatchEvent( changeEvent );
+		// Set to true to disable use of the keys
+		this.noKeys = false;
 
-			lastPosition.copy( this.object.position );
-			lastQuaternion.copy (this.object.quaternion );
+		// The four arrow keys
+		this.keys = { LEFT: 37, UP: 38, RIGHT: 39, BOTTOM: 40 };
 
-		}
+		// Mouse buttons
+		this.mouseButtons = { ORBIT: THREE.MOUSE.LEFT, ZOOM: THREE.MOUSE.MIDDLE, PAN: THREE.MOUSE.RIGHT };
 
-	};
+		////////////
+		// internals
 
+		var scope = this;
 
-	this.reset = function () {
+		var rotateStart = new THREE.Vector2();
+		var rotateEnd = new THREE.Vector2();
+		var rotateDelta = new THREE.Vector2();
 
-		state = STATE.NONE;
+		var panStart = new THREE.Vector2();
+		var panEnd = new THREE.Vector2();
+		var panDelta = new THREE.Vector2();
 
-		this.target.copy( this.target0 );
-		this.object.position.copy( this.position0 );
-		this.object.zoom = this.zoom0;
+		var dollyStart = new THREE.Vector2();
+		var dollyEnd = new THREE.Vector2();
+		var dollyDelta = new THREE.Vector2();
 
-		this.object.updateProjectionMatrix();
-		this.dispatchEvent( changeEvent );
+		var STATE = { NONE : -1, ROTATE : 0, DOLLY : 1, PAN : 2, TOUCH_ROTATE : 3, TOUCH_DOLLY : 4, TOUCH_PAN : 5 };
 
-		this.update();
+		var state = STATE.NONE;
 
-	};
+		// for reset
 
-	this.getPolarAngle = function () {
+		this.target0 = this.target.clone();
+		this.position0 = this.object.position.clone();
+		this.zoom0 = this.object.zoom;
 
-		return phi;
+		// events
 
-	};
+		var changeEvent = { type: 'change' };
+		var startEvent = { type: 'start' };
+		var endEvent = { type: 'end' };
 
-	this.getAzimuthalAngle = function () {
+		// pass in x,y of change desired in pixel space,
+		// right and down are positive
+		var _pan = this.pan;
+		this.pan = function ( deltaX, deltaY ) {
 
-		return theta
+			var element = scope.domElement === document ? scope.domElement.body : scope.domElement;
 
-	};
+			_pan.call( this, deltaX, deltaY, element.clientWidth, element.clientHeight );
 
-	function getAutoRotationAngle() {
+		};
 
-		return 2 * Math.PI / 60 / 60 * scope.autoRotateSpeed;
+		var _update = this.update;
```

Comment:
Yes, I can get rid of the inheritance. I'll just need to expose the constraint's `.minZoom`, `.maxDistance`, etc. via property on the controls, which is easy.

I'll also need to figure a way get rid of the last couple `dispatchEvent` in the constraint (because `this`, the event emitter, will not be the same).